src/mqtt_client.rs

```diff
@@ -725,7 +740,7 @@ impl<'buf, TcpStack: TcpClientStack, Clock: embedded_time::Clock> Minimq<'buf, T
     /// A `Minimq` object that can be used for publishing messages, subscribing to topics, and
     /// managing the MQTT state.
     pub fn new(
-        broker: IpAddr,
+        broker: impl Into<Broker>,
```

Comment:
Why not `Broker` and make the user do an explicit `into()`? That seems more common.

Reply:
I was trying to make it so users didn't have to change their API usage, but the type param forced it anyways. I'll revert this.